openquake/hazardlib/calc/gmf.py

```diff
@@ -76,23 +76,23 @@ class GmfComputer(object):
         case non-correlated ground motion fields are calculated.
         Correlation model is not used if ``truncation_level`` is zero.
     """
-    def __init__(self, rupture, sites, imts, gsim,
+    def __init__(self, rupture, sites, imts, gsims,
                  truncation_level=None, correlation_model=None):
         assert sites and imts, (sites, imts)
         self.rupture = rupture
         self.sites = sites
         self.imts = imts
-        self.gsim = gsim
+        self.gsims = gsims
         self.truncation_level = truncation_level
         self.correlation_model = correlation_model
-        self.ctx = gsim.make_contexts(sites, rupture)
+        self.ctx = {gsim: gsim.make_contexts(sites, rupture) for gsim in gsims}
```

Comment:
This is just a comment for a possible future improvement. Creating contexts for all the GMPEs is not efficient since in most of the cases we are recomputing the same rupture-site distances several times. It would be smarter to create a context which is the union of the context parameters needed by all the GMPEs used.